src/app/rmm/useragent.ts

```diff
@@ -0,0 +1,29 @@
+// --------- BEGIN RUNBOX LICENSE ---------
+// Copyright (C) 2016-2018 Runbox Solutions AS (runbox.com).
+// 
+// This file is part of Runbox 7.
+// 
+// Runbox 7 is free software: You can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at your
+// option) any later version.
+// 
+// Runbox 7 is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License
+// along with Runbox 7. If not, see <https://www.gnu.org/licenses/>.
+// ---------- END RUNBOX LICENSE ----------
+import { timeout } from 'rxjs/operators';
+import { HttpClient, HttpHeaders } from '@angular/common/http'
+import { RMM } from '../rmm';
+//ABSTRACT: this class has the http client that is used on this app
+export class UserAgent {
```

Comment:
I'm not sure this needs to exist: why not just have an `HttpClient` in the `RMM` class?

Reply:
Since the RMM class is simple OO, the useragent could be an instance of some other HttpClient. For example axios. This way other frameworks are able to re-use this same RMM class and just create an instance with their own UserAgent.. no need to have any angular deps to use the RMM class in an react project.

Reply:
The only class that ties RMM to angular is probably the useragent class which uses:

import { HttpClient, HttpHeaders } from '@angular/common/http'

The more RMM class is framework agnostic, the best i think...

Reply:
We're not being RMM agnostic here though: we're just putting our own box over Angular, and keeping its interface. If we were to move to axios tomorrow we'll still need to rewrite the entire RMM class at the very least, and since `load()` and some other methods (`create()`, `resend()` and `delete()` in Profile, and probably a lot more) “leak out” the underlying request object, in the end or entire codebase is still tightly coupled with Angular's HttpClient specifically, and not on any abstraction. And if you did actually build a different HTTP client that uses the same interface as HttpClient, you can swap it for the entire project just as well: since we're using Dependency Injection everywhere, we can provide anything we want as the HttpClient, [like we currently do in our unit tests](https://github.com/runbox/runbox7/blob/master/src/app/mailviewer/singlemailviewer.component.spec.ts#L79). That's the purpose of it being Injectable instead of something you `new` yourself: so that you can trivially substitute something else that implements the same interface.

Reply:
There is always room for improvement.

It would be cool to have that agnostic however this is not that important as long as it works inside runbox7 project